s3tests_boto3/functional/__init__.py

```diff
@@ -694,6 +694,10 @@ def get_tenant_aws_secret_key():
 def get_tenant_display_name():
     return config.tenant_display_name
 
+def get_tenant_name():
+    s = config.tenant_display_name
+    return s[:s.index('$')]
```

Comment:
this works for the default value from s3tests.conf.SAMPLE:
```ini
[s3 tenant]
# tenant display_name set in vstart.sh
display_name = testx$tenanteduser
```
but fails in teuthology where [qa/tasks/s3tests.py](https://github.com/ceph/ceph/blob/53b533f/qa/tasks/s3tests.py#L82) sets display_name to `Mr.testx$tenanteduser`. rgw sees these requests as:
```
REQUEST_URI=/Mr.testx%3Atest-fak34zuywqcog5fhyh4e22bc-475?encoding-type=url
```
and rejects it with `400 InvalidTenantName` because of the `.` in `Mr.testx`

Reply:
since we need the tenant name for this test new case, i added a new `tenant` variable to the `[s3 tenant]` config section. i added a new commit to https://github.com/ceph/ceph/pull/59169 that configures this in teuthology